plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/content/ConstraintViewConstraintColumnLabelProvider.java

```diff
@@ -105,12 +106,15 @@ public Image getImage(Object element) {
 				return getColoredCircleImage(color);
 			}
 
-			final ConstraintProperties constraintProperties =
-				controller.getFeatureModelManager().getVariableFormula().getAnalyzer().getAnalysesCollection().getConstraintProperty(constraint);
-			if (constraintProperties.hasStatus(ConstraintProperties.ConstraintStatus.REDUNDANT)) {
-				return GUIDefaults.FM_INFO;
-			}
+			final Boolean isAutomaticCalculation = FeatureModelProperty.getBooleanProperty(controller.getFeatureModelManager().getSnapshot().getProperty(),
+					FeatureModelProperty.TYPE_CALCULATIONS, FeatureModelProperty.PROPERTY_CALCULATIONS_RUN_AUTOMATICALLY);
+			if ((isAutomaticCalculation != null) && isAutomaticCalculation) {
 
+				final ConstraintProperties constraintProperties = controller.getAnalyzer().getAnalysesCollection().getConstraintProperty(constraint);
```

Comment:
As this is the same condition, it would be easier and safer if we just check whether the analyzer is null.
```
final FeatureModelAnalyzer analyzer = controller.getAnalyzer();
if (analyzer != null) {
...
```